shared/src/main/scala/drt/shared/Api.scala

```diff
@@ -46,14 +46,18 @@ object MilliDate {
 }
 
 object FlightParsing {
-  val iataRe: Regex = """([A-Z0-9]{2})(\d{1,4})(\w)?""".r
-  val icaoRe: Regex = """([A-Z]{2,3})(\d{1,4})(\w)?""".r
+  val iataRe: Regex = "^([A-Z0-9]{2}?)([0-9]{1,4})([A-Z]*)$".r
+  val icaoRe: Regex = "^([A-Z]{2,3}?)([0-9]{1,4})([A-Z]*)$".r
 
-  def parseIataToCarrierCodeVoyageNumber(iata: String): Option[(String, String)] = {
+  def flightCodeToParts(iata: String): (CarrierCode, VoyageNumberLike, Option[ArrivalSuffix]) = {
     iata match {
-      case iataRe(carriercode, voyageNumber, _) => Option((carriercode, voyageNumber))
-      case icaoRe(carriercode, voyageNumber, _) => Option((carriercode, voyageNumber))
-      case what => None
+      case iataRe(cc, vn, suffix) =>
+        val carrierCode = CarrierCode(cc)
+        val voyageNumber = VoyageNumber(vn)
+        val arrivalSuffix = if (suffix.nonEmpty) Option(ArrivalSuffix(suffix)) else None
```

Comment:
you extract this if to a function as you are doing it twice.

Reply:
oops i totally missed that